chat_server/config/redis_config.ts

```diff
@@ -0,0 +1,23 @@
+import dotenv from "dotenv";
+import { createClient } from "redis";
+
+dotenv.config();
+
+const isDev = !process.env.DOCKER_HOST_IP;
+
+const redisClient = createClient({
+	url: isDev
+		? "redis://localhost:6379"
+		: `${process.env.DOCKER_HOST_IP}:${process.env.REDIS_PORT}`,
+});
+
+const connectRedis = async () => {
+	try {
+		await redisClient.connect();
+		console.log("Connected to Redis");
+	} catch (error) {
+		console.error("Could not connect to Redis", error);
+	}
+};
+
+export { redisClient, connectRedis };
```

Comment:
@hojun3377 

`nit` Redis 클라이언트가 여러 모듈에서 사용될 경우 `싱글톤`으로 관리하는 패턴이 좋습니다.
연결 실패 시 retry도 추가해주면 더 안정적인 Redis 운영이 가능할 것 같습니다.

```suggestion
import dotenv from "dotenv";
import { createClient } from "redis";

dotenv.config();

class RedisSingleton {
	private static instance: ReturnType<typeof createClient> | null = null;

	private constructor() {}

	static getInstance() {
		if (!this.instance) {
			this.instance = createClient({
				url: process.env.REDIS_URL || "redis://localhost:6379",
			});
			this.instance.on("error", (err) => console.error("Redis Client Error", err));
		}
		return this.instance;
	}
}

const redisClient = RedisSingleton.getInstance();

const connectRedis = async () => {
	try {
		await redisClient.connect();
		console.log("Connected to Redis");
	} catch (error) {
		console.error("Could not connect to Redis, retry...", error);
		setTimeout(connectRedis, 5000); // 5초 후 재시도
	}
};

export { redisClient, connectRedis };
```